### PR TITLE
Whole genome chgpt dmr

### PR DESCRIPTION
**Pull request by tcairns-soundag**
Updates to the changepoint function so that it can work across a whole genome (in chunks)
Updated the changepoint figure to be a better representation of the changepoints that exist

### CODE REVIEW

File: R/Sound_DMR.R [hunk 1]
```diff
@@ -974,74 +974,89 @@ changepoint_analysis <- function(whole_df,
...
+             group_num = (row_num %/% subset_size) + 1)
+    subset_col <- "group_num"
+    target_genes <- unique(whole_df$group_num)
+  }
+  
+  subsets <- unique(whole_df[[subset_col]])
+  large_genes <- c()
+  
+  # Create the structure of the final df
   everything <- tibble::as_tibble(matrix(ncol = 8))
```
Comment (kprabhu-soundag) [inline R/Sound_DMR.R:996]: nit : I think we can change the name the variable 'everything' here.

Comment (tcairns-soundag) [reply R/Sound_DMR.R:1060]: For sure!

**Comment by jcolicchio-soundag** [comment]
Looks good to me!

**Review by jcolicchio-soundag — APPROVED** [review]
Looks great!

File: R/Sound_DMR.R [hunk 2]
```diff
@@ -940,13 +989,28 @@ plot_changepoints <- function(data, changepoint_obj, gene_name, penalty_val,
...
     return(plot)
   }, error = function(e) {
-    return(paste('Figure could not be created for:', gene_name))
+    return(warning('Figure could not be created for:', gene_name))
   })
 }
 
+#' Save Plot
+#' 
+#' Save a plot
```
Comment (mflashman-soundag) [inline R/Sound_DMR.R:1001]: nit: rm

File: R/Sound_DMR.R [hunk 3]
```diff
@@ -963,6 +1027,9 @@ plot_changepoints <- function(data, changepoint_obj, gene_name, penalty_val,
 #' @param z_col (str) - the column to run the changepoint analysis on. This
 #' can be any column, but for DMR analysis we recommend using the z scores for
 #' a fixed effect variable.
+#' @param whole_genome (boolean) - whether to run changepoints on a whole genome
```
Comment (mflashman-soundag) [inline R/Sound_DMR.R:1030]: nit: Could also make this parameter `subset_by="Gene" or "Chromosome"` (or any other column for that matter )

**Review by mflashman-soundag — APPROVED** [review]
lgtm

**Comment by jcolicchio-soundag** [comment]
can we squash and merge?

**Pull request merged**